.gitlab-ci-github-status-updates.yml

```diff
@@ -0,0 +1,38 @@
+variables:
+  GITHUB_AUTH: "Authorization: bearer $GITHUB_BOT_TOKEN"
+
+  GITHUB_STATUS_API_URL: "https://api.github.com/repos/mendersoftware/$CI_PROJECT_NAME/statuses/$CI_COMMIT_SHA"
+  GITHUB_STATUS_API_JSON_F: '{"state": "%s", "context": "ci/gitlab", "target_url": "%s", "description": "%s"}'
+
+stages:
+  # These have no effect and are only for documentation purposes. .pre and .post
+  # stages run at the very start and very end of a pipeline, regardless of where
+  # they are defined.
+  - .pre
+  - .post
+
+github:start:
+  stage: .pre
+  # Keep overhead low by using a small image with curl preinstalled.
+  image: appropriate/curl
+  script:
+    - 'JSON="$(printf "$GITHUB_STATUS_API_JSON_F" pending "$CI_PIPELINE_URL" "Pipeline running on Gitlab")"'
+    - 'curl -f -H "$GITHUB_AUTH" -d "$JSON" "$GITHUB_STATUS_API_URL"'
```

Comment:
Why are these single quoted? I suspect this will teach me something...

Reply:
I can't say with certainty at this point that they all need to be quoted, but I got a lot of parsing errors in the beginning because of the mix of the `$()` notation and variables that have spaces in them. In the end I decided to just be consistent and use single quotes in every line that might have variables with spaces, which will let the shell expand them instead.